cmd/busmanager/main.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.einride.tech/can/pkg/socketcan"
+	"go.uber.org/zap"
+
+	"github.com/macformula/hil/canlink"
+)
+
+type Handler struct {
+	name string
+}
+
+func NewHandler() *Handler {
+	handler := &Handler{
+		name: "testHandler",
+	}
+
+	return handler
+}
+
+func (h *Handler) Handle(
+	broadcast chan canlink.TimestampedFrame,
+	transmit chan canlink.TimestampedFrame,
+) {
+	go func() {
+		for {
+			select {
+			case frame := <-broadcast:
+				fmt.Println("RECEIVED: ", frame.Frame)
+			default:
+			}
+		}
+	}()
+
+	go func() {
+		var i byte
+
+		for {
+			time.Sleep(2 * time.Millisecond)
+
+			frame := canlink.TimestampedFrame{}
+			copy(frame.Frame.Data[:], []byte{i})
+			frame.Time = time.Now()
+
+			i = i + 1
+
+			transmit <- frame
+		}
+	}()
+}
+
+func (h *Handler) Name() string {
+	return "Handler"
+}
+
+func main() {
+	ctx := context.Background()
+
+	loggerConfig := zap.NewDevelopmentConfig()
+	logger, err := loggerConfig.Build()
+
+	conn, err := socketcan.DialContext(context.Background(), "can", "vcan0")
```

Comment:
does it make sense to include this code into a BusManager function? I think it would be nice to be able to call something like
```
manager =canlink.BusManager("vcan0")
```
Same with the logger. If every BusManager needs a logger, can it create its own during construction?

Reply:
There may be other objects that interact with can and this way they can all share the same socket connection. What we should do is have a look at the dialcontext implementation. If this creates a new connection on each call then it should remain the same. But if it just returns an instance of a pre-existing connection it should be fine to make the change.

This is how zap logger works as well. You create a logger instance at the top level and pass it in to all objects that will perform logging. I understand this is a little different from how python handles things.

Reply:
Yes, logger should be injected. However for the connection, it might make more sense for the `BusManager` to own it. I'll look into `DialContext`.